tszip/compression.py

```diff
@@ -192,9 +192,9 @@ def compress_zarr(ts, root, variants_only=False):
     # Sequence length is stored as an attr for compatibility with older versions of tszip
     del columns["sequence_length"]
 
-    # Schemas and metadata need to be converted to arrays
+    # Schemas, metadata and units need to be converted to arrays
     for name in columns:
-        if name.endswith("metadata_schema"):
+        if name.endswith("metadata_schema") or name == "time_units":
```

Comment:
Would ``if isinstance(column, str)`` work here? Is the ``metadata`` column below also effectively assuming it's a ``bytes`` value?

Reply:
I've pushed a commit that does that on compress. On decompress we can't detect, although the best idea might be to write the type to the zarr metadata. Will try that later.